tests/functional/ephemeral/test_ephemeral.py

```diff
@@ -0,0 +1,216 @@
+
+with source_data as (
+
+    select 1 as id
+    union all
+    select null as id
+
+)
+
+select *
+from source_data
+
+"""
+
+
+@pytest.fixture(scope="class", autouse=True)
+def setUp(project):
+    project.run_sql_file(project.test_data_dir / Path("seed.sql"))
+
+
+class TestEphemeralMulti:
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "dependent.sql": models__dependent_sql,
+            "double_dependent.sql": models__double_dependent_sql,
+            "super_dependent.sql": models__super_dependent_sql,
+            "base": {
+                "female_only.sql": models__base__female_only_sql,
+                "base.sql": models__base__base_sql,
+                "base_copy.sql": models__base__base_copy_sql,
+            },
+        }
+
+    def test_ephemeral_multi(self, project):
+        results = run_dbt(["run"])
+        assert len(results) == 3
+
+        check_relations_equal(project.adapter, ["seed", "dependent"])
+        check_relations_equal(project.adapter, ["seed", "double_dependent"])
+        check_relations_equal(project.adapter, ["seed", "super_dependent"])
+        assert os.path.exists("./target/run/test/models/double_dependent.sql")
+        with open("./target/run/test/models/double_dependent.sql", "r") as fp:
+            sql_file = fp.read()
+
+        sql_file = re.sub(r"\d+", "", sql_file)
+        expected_sql = (
+            'create view "dbt"."test_test_ephemeral"."double_dependent__dbt_tmp" as ('
+            "with __dbt__cte__base as ("
+            "select * from test_test_ephemeral.seed"
+            "),  __dbt__cte__base_copy as ("
+            "select * from __dbt__cte__base"
+            ")-- base_copy just pulls from base. Make sure the listed"
+            "-- graph of CTEs all share the same dbt_cte__base cte"
+            "select * from __dbt__cte__base where gender = 'Male'"
+            "union all"
+            "select * from __dbt__cte__base_copy where gender = 'Female'"
+            ");"
+        )
+        sql_file = "".join(sql_file.split())
+        expected_sql = "".join(expected_sql.split())
+        assert sql_file == expected_sql
+
+
+class TestEphemeralNested:
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "ephemeral_level_two.sql": models_n__ephemeral_level_two_sql,
+            "root_view.sql": models_n__root_view_sql,
+            "ephemeral.sql": models_n__ephemeral_sql,
+            "source_table.sql": models_n__source_table_sql,
+        }
+
+    def test_ephemeral_nested(self, project):
+        results = run_dbt(["run"])
+        assert len(results) == 2
+        assert os.path.exists("./target/run/test/models/root_view.sql")
+        with open("./target/run/test/models/root_view.sql", "r") as fp:
+            sql_file = fp.read()
+
+        sql_file = re.sub(r"\d+", "", sql_file)
+        expected_sql = (
+            'create view "dbt"."test_test_ephemeral"."root_view__dbt_tmp" as ('
```

Comment:
is renaming the hardcoded value for `test_ephemeral_` -> `test_test_ephemeral` the best way to handle file name changes here or is there something else we might prefer too?